contrib/git-sync

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+
+"""
+Push git repostory to remote modifying the remote working directory.
+"""
+
+import os
+import sys
+import uuid
+import subprocess
+from urllib.parse import urlparse
+
+
+def run(cmd):
+    return subprocess.check_output(cmd).decode().strip()
+
+
+if len(sys.argv) < 2:
+    print("Usage: git-sync REMOTE [BRANCH]")
+    sys.exit(2)
+
+remote = sys.argv[1]
+
+if len(sys.argv) > 2:
+    branch = sys.argv[2]
+else:
+    branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"])
+
+tmp_branch = "tmp-%s" % uuid.uuid4()
+
+remote_url = urlparse(run(["git", "remote", "get-url", remote]))
```

Comment:
Yes, of course I added remote for the host I'm pushing to, this is the way to push
stuff in git.

Here is an example from my local repo:

```
$ git remote -v
host4	nsoffer@host4:src/ovirt-imageio/.git (fetch)
host4	nsoffer@host4:src/ovirt-imageio/.git (push)
```

I will describe the entire flow in the module docstring.

Reply:
It is a good trick to leverage git to handle your vms as remotes, but yes, this would need to be documented, it is not so obvious.

As a possible alternative, what about use a config file? Since the script has the name of a git plugin, it could have its own configuration in the hidden git folder of the repo, something like `./.git/.gitsync-conf`. For instance, with yaml it could look like:
```yaml
host4: 
  hostname: host4
  path: src/ovirt-imageio
  username: nsoffer
```
And it would be parsed by the python yaml library as a dictionary. And then do:
```python
    with open(config, 'r', encoding='utf-8') as conf_file:
        loaded_conf = yaml.safe_load(conf_file)
    config = loaded_conf[remote]
    netlock = f"{config["username"]}@{config["hostname"]}"
    path = config["path"]
```

Reply:
Nevermind the alternative, you push force to the remote directly, so it needs to be kept as a remote.

Reply:
The git url is not what you see in ‘git remote -v’ - it return ‘ssh://‘ and parsing it return what we need.

Reply:
Right, like that it works fine :)